lib/db2.js

```diff
@@ -50,6 +51,8 @@ function DB2(settings) {
     ';PORT=' + this.portnumber +
     ';PROTOCOL=' + this.protocol;
 
+  this.connStr = this.dsn || this.connStrGenerate;
+
   this.schema = this.username;
   if (settings.schema) {
     this.connStr += ';CurrentSchema=' + settings.schema;
```

Comment:
This should not be applied when `settings.dsn` is specified, as `settings.dsn` should already include Schema if necessary.